spyder/plugins/editor/plugin.py

```diff
@@ -2872,7 +2872,9 @@ def setup_open_files(self):
             # spyder-ide/spyder#8458.
             if layout:
                 is_vertical, cfname, clines = layout.get('splitsettings')[0]
-                if cfname in filenames:
+                # Check that a value for current line exist for each filename
+                # in the available settings. See spyder-ide/spyder#12201
+                if cfname in filenames and len(filenames) == len(clines):

```

Comment:
@dalthviz, what happens if this new verification you added doesn't pass? It seems no files will be opened?

Reply:
If the verification doesn't pass it will lead to the opening of the files following the order of the `filenames` list (it will enter the `else` statement), without opening them in the stored line position and ignoring the current filename stored in the layout settings:

https://github.com/spyder-ide/spyder/blob/680b8606187192db7894185b23562cd4cd2dc559/spyder/plugins/editor/plugin.py#L2898-L2902

The `load` method also handles the case where `len(filenames) != len(clines)` (over there `clines=goto`):

https://github.com/spyder-ide/spyder/blob/680b8606187192db7894185b23562cd4cd2dc559/spyder/plugins/editor/plugin.py#L1955-L1956

Reply:
Great!! Thanks for the explanation @dalthviz!